lib/inc/hocon/config_value.hpp

```diff
@@ -52,6 +45,25 @@ namespace hocon {
         friend class config_concatenation;
     public:
         /**
+         * The type of a configuration value (following the <a
+         * href="http://json.org">JSON</a> type schema).
+         */
+        enum class type {
+            OBJECT, LIST, NUMBER, BOOLEAN, CONFIG_NULL, STRING, UNSPECIFIED
+        };
+        static char const* type_name(type t) {
```

Comment:
Ah, that sounds like a bug in the compiler (the switch covers all valid enum values for the enum class). I'll add a default section anyway.